### PR TITLE
Add a dropdown with the unit of the max upload size

### PR DESCRIPTION
**Pull request by nickvergessen**
Replaces #13048 

@LukasReschke @DeepDiver1975 @MorrisJobke

### CODE REVIEW

File: apps/files/admin.php [hunk 1]
```diff
@@ -22,26 +22,34 @@
...
-$maxUploadFilesize = OCP\Util::humanFileSize(min($upload_max_filesize, $post_max_size));
-if($_POST && OC_Util::isCallRegistered()) {
-	if(isset($_POST['maxUploadSize'])) {
-		if(($setMaxSize = OC_Files::setUploadLimit(OCP\Util::computerFileSize($_POST['maxUploadSize']))) !== false) {
-			$maxUploadFilesize = OCP\Util::humanFileSize($setMaxSize);
+$phpIni = new \bantu\IniGetWrapper\IniGetWrapper();
+$maxUploadFileSize = min($phpIni->getBytes('upload_max_filesize'), $phpIni->getBytes('post_max_size'));
+if ($_POST && OC_Util::isCallRegistered()) {
+	if (isset($_POST['maxUploadSize'])) {
+		if (($setMaxSize = OC_Files::setUploadLimit(OCP\Util::computerFileSize($_POST['maxUploadSize']))) !== false) {
```
Comment (LukasReschke) [inline apps/files/admin.php]: Please add an explicit typecase for `maxUploadSize` - i.e. `(int)` or `(string)`

Comment (nickvergessen) [inline apps/files/admin.php]: fixed

File: apps/files/admin.php [hunk 2]
```diff
@@ -22,26 +22,35 @@
 */
 
 OCP\User::checkAdminUser();
+$l = \OC_L10N::get('files');
 
-$htaccessWorking=(getenv('htaccessWorking')=='true');
+$htaccessWorking = getenv('htaccessWorking') == 'true';
+$uploadChangable = $htaccessWorking && is_writable(OC::$SERVERROOT.'/.htaccess');
+$maxFileSizeUnits = [
+	'B' => $l->t('B'),
```
Comment (DeepDiver1975) [inline apps/files/admin.php]: translating these - seriously?

MB will be MB - right?

Comment (nickvergessen) [inline apps/files/admin.php]: for normal europeans yes, but in other languages not, russian e.g. uses `МБ` for `MB`, but as you command

File: apps/files/admin.php [hunk 3]
```diff
@@ -22,26 +22,35 @@
...
-$upload_max_filesize = OCP\Util::computerFileSize(ini_get('upload_max_filesize'));
-$post_max_size = OCP\Util::computerFileSize(ini_get('post_max_size'));
-$maxUploadFilesize = OCP\Util::humanFileSize(min($upload_max_filesize, $post_max_size));
-if($_POST && OC_Util::isCallRegistered()) {
-	if(isset($_POST['maxUploadSize'])) {
-		if(($setMaxSize = OC_Files::setUploadLimit(OCP\Util::computerFileSize($_POST['maxUploadSize']))) !== false) {
-			$maxUploadFilesize = OCP\Util::humanFileSize($setMaxSize);
-		}
+$phpIni = new \bantu\IniGetWrapper\IniGetWrapper();
+$maxUploadFileSize = min($phpIni->getBytes('upload_max_filesize'), $phpIni->getBytes('post_max_size'));
```
Comment (nickvergessen) [inline apps/files/admin.php]: @bantu @DeepDiver1975 I just saw, you had a discussion about this in
https://github.com/owncloud/core/pull/11131/files#r17835141
mind voting again in here?

File: apps/files/admin.php [hunk 4]
```diff
@@ -22,26 +22,35 @@
...
 
-$upload_max_filesize = OCP\Util::computerFileSize(ini_get('upload_max_filesize'));
-$post_max_size = OCP\Util::computerFileSize(ini_get('post_max_size'));
-$maxUploadFilesize = OCP\Util::humanFileSize(min($upload_max_filesize, $post_max_size));
-if($_POST && OC_Util::isCallRegistered()) {
-	if(isset($_POST['maxUploadSize'])) {
-		if(($setMaxSize = OC_Files::setUploadLimit(OCP\Util::computerFileSize($_POST['maxUploadSize']))) !== false) {
-			$maxUploadFilesize = OCP\Util::humanFileSize($setMaxSize);
-		}
+$phpIni = new \bantu\IniGetWrapper\IniGetWrapper();
```
Comment (nickvergessen) [inline apps/files/admin.php]: Should be replaced with `\OC::$server->getIniWrapper()`

File: apps/files/admin.php (hunk 3 above)
Comment (DeepDiver1975) [inline apps/files/admin.php]: I'm fine with this

**Comment by DeepDiver1975** [comment]
rescheduling this for 8.2 - we are entering freeze for 8.1 tomorrow

**Comment by nickvergessen** [comment]
Rebased and addressed the comment.

Also mentioning @jancborchardt for UI

**Comment by MorrisJobke** [comment]
This was fixed in my commit:

Before:

bildschirmfoto von 2015-07-03 10-03-59

After:

bildschirmfoto von 2015-07-03 10-04-19

**Comment by nickvergessen** [comment]
This was just caused by a missing change on my rebase, going to push the fix when jenkins reports back

**Comment by nickvergessen** [comment]
@owncloud-bot retest this please

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

**Comment by DeepDiver1975** [comment]
this one is obsolete once we move to webdav 12353

**Comment by PVince81** [comment]
Yes and no. We can get the quota / free space from a PROPFIND but not the max upload size from PHP. That one would have to be provided through "oc.js" or other means and computed in the UI.

Or are you saying that PUT doesn't have any max upload limitation ?

**Comment by DeepDiver1975** [comment]
> Or are you saying that PUT doesn't have any max upload limitation ?

yes ;-)

**Comment by MorrisJobke** [comment]
> this one is obsolete once we move to webdav #12353

I talked to @nickvergessen and we don't want to put any effort into this anymore because it will be obsolete soon -> close

**Pull request closed without merging**